internal/lang/marks/marks.go

```diff
@@ -51,3 +93,22 @@ const Ephemeral = valueMark("Ephemeral")
 // another value's type. This is part of the implementation of the console-only
 // `type` function.
 const TypeType = valueMark("TypeType")
+
+type DeprecationMark struct {
+	Message string
+	Origin  *hcl.Range
```

Comment:
Should we try tracking some source address and attribute path info here, maybe something like a `globalref.Reference`? While we can render the source position once the diagnostic is printed, that doesn't directly tell the user which type and attribute path is being used.